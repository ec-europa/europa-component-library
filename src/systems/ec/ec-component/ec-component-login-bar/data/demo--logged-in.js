module.exports = {
  logged_in: true,
  message: 'Logged in as Jane Doe',
  link: {
    href: '../../example.html#',
    label: 'Log out',
  },
};
