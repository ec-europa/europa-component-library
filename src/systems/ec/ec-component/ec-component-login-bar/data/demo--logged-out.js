module.exports = {
  logged_in: false,
  link: {
    href: '../../example.html#',
    label: 'Log in',
  },
};
