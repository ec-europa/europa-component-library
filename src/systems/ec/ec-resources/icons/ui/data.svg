<!-- Generated by IcoMoon.io -->
<svg version="1.1" xmlns="http://www.w3.org/2000/svg" width="28" height="32" viewBox="0 0 28 32">
<title>data</title>
<path d="M14.075 0.376c-7.791 0-14.075 3.078-14.075 7.022v18.115c0 3.848 6.284 7.022 14.075 7.022s14.043-3.11 14.043-6.958v-18.18c0-3.944-6.38-7.022-14.043-7.022zM25.137 25.513c0 1.218-3.751 4.008-11.062 4.008-7.214 0-11.062-2.789-11.062-4.008v-4.713c2.533 1.603 6.541 2.725 11.062 2.725 4.393 0 8.432-1.026 11.062-2.725v4.713zM25.137 16.503c0 1.218-3.751 4.040-11.062 4.040-7.214 0-11.062-2.822-11.062-4.040v-4.681c2.533 1.571 6.541 2.693 11.062 2.693 4.393 0 8.432-1.026 11.062-2.693v4.681zM14.075 11.437c-7.214 0-11.062-2.822-11.062-4.040s3.751-4.008 11.062-4.008c7.214 0 11.062 2.789 11.062 4.008s-3.848 4.040-11.062 4.040z"></path>
</svg>
