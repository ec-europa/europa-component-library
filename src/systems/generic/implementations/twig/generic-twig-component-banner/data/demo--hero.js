module.exports = {
  type: 'hero',
  image: '../../assets/demo_images/nature-demo-1.jpg',
  title: 'White Paper on the future of Europe',
  href: '../../example.html#hero-banner',
  description:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation.',
};
