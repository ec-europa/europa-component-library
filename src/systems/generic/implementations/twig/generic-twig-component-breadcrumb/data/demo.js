module.exports = function context(system) {
  const c = {
    segments: [
      {
        href: '../../example.html#',
        label: 'Home',
      },
      {
        href: '../../example.html#',
        label: 'Priorities',
      },
      {
        href: '../../example.html#',
        label: 'Category',
      },
      {
        href: '../../example.html#',
        label: 'Jobs, Growth and Investment',
      },
      {
        label: 'Page title',
      },
    ],
    icon: {
      icon_path: `../../${system}-preset-website/images/icons/symbol-defs.svg`,
      icon: 'right',
      size: 'xxs',
    },
    _demo: {
      scripts: `
        document.addEventListener('DOMContentLoaded', function () { 
          ECL.initBreadcrumb();
        });`,
    },
  };

  return c;
};
