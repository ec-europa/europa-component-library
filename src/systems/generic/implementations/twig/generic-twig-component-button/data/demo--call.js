module.exports = function context(system) {
  const c = {
    label: 'Call-to-action button',
    variant: 'call',
    icon: {
      icon_path: `../../${system}-preset-website/images/icons/symbol-defs.svg`,
      icon: 'right',
      size: 'xs',
    },
  };

  return c;
};
