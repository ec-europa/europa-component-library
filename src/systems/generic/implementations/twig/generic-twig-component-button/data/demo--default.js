module.exports = {
  label: 'Default',
  modifier: 'default',
};
