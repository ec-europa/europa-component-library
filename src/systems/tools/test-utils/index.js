const merge = require('deepmerge');
const twing = require('@ecl-twig/ec-storybook/.storybook/environment');

const renderTwigFileAsNode = (file, options) =>
  new Promise((resolve, reject) => {
    try {
      const html = twing.render(file, options);
      const test = document.createElement('test');
      test.innerHTML = html.trim();
      resolve(test);
    } catch (error) {
      reject(error);
    }
  });

module.exports = {
  merge,
  renderTwigFileAsNode,
};
