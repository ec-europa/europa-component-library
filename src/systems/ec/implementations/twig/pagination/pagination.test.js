import { merge, renderTwigFileAsNode } from '@ecl-twig/test-utils';

// Import data for tests
import data from './demo/data';

describe('EC - Pagination', () => {
  const template = '@ecl-twig/ec-component-pagination/ecl-pagination.html.twig';
  const render = (params) => renderTwigFileAsNode(template, params);

  describe('Default', () => {
    test('renders correctly', () => {
      expect.assertions(1);

      return expect(render(data)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra class names', () => {
      expect.assertions(1);

      const withExtraClasses = merge(data, {
        extra_classes: 'custom-class custom-class--test',
      });

      return expect(render(withExtraClasses)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra attributes', () => {
      expect.assertions(1);

      const withExtraAttributes = merge(data, {
        extra_attributes: [
          { name: 'data-test', value: 'data-test-value' },
          { name: 'data-test-1', value: 'data-test-value-1' },
        ],
      });

      return expect(render(withExtraAttributes)).resolves.toMatchSnapshot();
    });

    test('With missing input data returns the right warning message', () => {
      expect.assertions(1);

      const dataCompliance = { ...data, _compliance_: true };
      dataCompliance.label = '';
      dataCompliance.items[0].type = '';

      return expect(render(dataCompliance)).resolves.toMatchSnapshot();
    });
  });
});
