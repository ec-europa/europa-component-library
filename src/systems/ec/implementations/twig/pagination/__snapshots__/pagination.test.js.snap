// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EC - Pagination Default With missing input data returns the right warning message 1`] = `
<test>
  <nav
    aria-label=""
    class="ecl-pagination"
  >
    <ul
      class="ecl-pagination__list"
    >
      <li
        class="ecl-pagination__item"
      >
        <a
          aria-label="Go to previous page"
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-pagination__link"
          href="/example"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs ecl-icon--rotate-270 ecl-link__icon"
            focusable="false"
          >
            <use
              xlink:href="/icons.svg#ui--corner-arrow"
            />
          </svg>
          <span
            class="ecl-u-type-color-green ecl-compliance-inner ecl-compliance-success ecl-u-ma-2xs"
            title="[icon] - ecl compliant"
          >
            <svg
              aria-hidden="true"
              class="ecl-icon ecl-icon--xs"
              focusable="false"
            >
              <use
                xlink:href="/icons.svg#notifications--success"
              />
            </svg>
          </span>
           
          <span
            class="ecl-link__label"
          >
            Previous
          </span>
        </a>
        <span
          class="ecl-u-type-color-green ecl-compliance-inner ecl-compliance-success ecl-u-ma-2xs"
          title="[link--standalone] - ecl compliant"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs"
            focusable="false"
          >
            <use
              xlink:href="/icons.svg#notifications--success"
            />
          </svg>
        </span>
      </li>
      <li
        class="ecl-pagination__item"
      >
        <a
          aria-label="Go to page 24"
          class="ecl-link ecl-link--standalone ecl-pagination__link"
          href="/example"
        >
          <span
            class="ecl-link__label"
          >
            24
          </span>
        </a>
        <span
          class="ecl-u-type-color-green ecl-compliance-inner ecl-compliance-success ecl-u-ma-2xs"
          title="[link--standalone] - ecl compliant"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs"
            focusable="false"
          >
            <use
              xlink:href="/icons.svg#notifications--success"
            />
          </svg>
        </span>
      </li>
      <li
        class="ecl-pagination__item"
      >
        <a
          aria-label="Go to page 25"
          class="ecl-link ecl-link--standalone ecl-pagination__link"
          href="/example"
        >
          <span
            class="ecl-link__label"
          >
            25
          </span>
        </a>
        <span
          class="ecl-u-type-color-green ecl-compliance-inner ecl-compliance-success ecl-u-ma-2xs"
          title="[link--standalone] - ecl compliant"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs"
            focusable="false"
          >
            <use
              xlink:href="/icons.svg#notifications--success"
            />
          </svg>
        </span>
      </li>
      <li
        class="ecl-pagination__item ecl-pagination__item--current"
      >
        <span
          aria-current="true"
          aria-label="Page 26"
          class="ecl-pagination__text ecl-pagination__text--summary"
        >
          26
        </span>
        <span
          aria-current="true"
          class="ecl-pagination__text ecl-pagination__text--full"
        >
          Page 26
        </span>
      </li>
      <li
        class="ecl-pagination__item"
      >
        <a
          aria-label="Go to page 27"
          class="ecl-link ecl-link--standalone ecl-pagination__link"
          href="/example"
        >
          <span
            class="ecl-link__label"
          >
            27
          </span>
        </a>
        <span
          class="ecl-u-type-color-green ecl-compliance-inner ecl-compliance-success ecl-u-ma-2xs"
          title="[link--standalone] - ecl compliant"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs"
            focusable="false"
          >
            <use
              xlink:href="/icons.svg#notifications--success"
            />
          </svg>
        </span>
      </li>
      <li
        class="ecl-pagination__item"
      >
        <a
          aria-label="Go to page 28"
          class="ecl-link ecl-link--standalone ecl-pagination__link"
          href="/example"
        >
          <span
            class="ecl-link__label"
          >
            28
          </span>
        </a>
        <span
          class="ecl-u-type-color-green ecl-compliance-inner ecl-compliance-success ecl-u-ma-2xs"
          title="[link--standalone] - ecl compliant"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs"
            focusable="false"
          >
            <use
              xlink:href="/icons.svg#notifications--success"
            />
          </svg>
        </span>
      </li>
      <li
        class="ecl-pagination__item ecl-pagination__item--next"
      >
        <a
          aria-label="Go to next page"
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-after ecl-pagination__link"
          href="/example"
        >
          <span
            class="ecl-link__label"
          >
            Next
          </span>
           
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs ecl-icon--rotate-90 ecl-link__icon"
            focusable="false"
          >
            <use
              xlink:href="/icons.svg#ui--corner-arrow"
            />
          </svg>
          <span
            class="ecl-u-type-color-green ecl-compliance-inner ecl-compliance-success ecl-u-ma-2xs"
            title="[icon] - ecl compliant"
          >
            <svg
              aria-hidden="true"
              class="ecl-icon ecl-icon--xs"
              focusable="false"
            >
              <use
                xlink:href="/icons.svg#notifications--success"
              />
            </svg>
          </span>
        </a>
        <span
          class="ecl-u-type-color-green ecl-compliance-inner ecl-compliance-success ecl-u-ma-2xs"
          title="[link--standalone] - ecl compliant"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs"
            focusable="false"
          >
            <use
              xlink:href="/icons.svg#notifications--success"
            />
          </svg>
        </span>
      </li>
    </ul>
  </nav>
  <!-- ecl2-compliance: pagination -->
  <div
    class="ecl2-compliance ecl-u-border-width-1 ecl-u-border-color-red ecl-u-border-style-solid ecl-u-ma-xs ecl-u-pr-s ecl-u-bg-white"
  >
    <p
      class="ecl-u-type-color-grey ecl-u-ml-m ecl-u-type-xs"
    >
      [pagination] -  {{ label }} needs to set and filled in for accessibility reasons. Previous link is mandatory.
    </p>
  </div>
</test>
`;

exports[`EC - Pagination Default renders correctly 1`] = `
<test>
  <nav
    aria-label="Pagination"
    class="ecl-pagination"
  >
    <ul
      class="ecl-pagination__list"
    >
      <li
        class="ecl-pagination__item ecl-pagination__item--previous"
      >
        <a
          aria-label="Go to previous page"
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-pagination__link"
          href="/example"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs ecl-icon--rotate-270 ecl-link__icon"
            focusable="false"
          >
            <use
              xlink:href="/icons.svg#ui--corner-arrow"
            />
          </svg>
           
          <span
            class="ecl-link__label"
          >
            Previous
          </span>
        </a>
      </li>
      <li
        class="ecl-pagination__item"
      >
        <a
          aria-label="Go to page 24"
          class="ecl-link ecl-link--standalone ecl-pagination__link"
          href="/example"
        >
          <span
            class="ecl-link__label"
          >
            24
          </span>
        </a>
      </li>
      <li
        class="ecl-pagination__item"
      >
        <a
          aria-label="Go to page 25"
          class="ecl-link ecl-link--standalone ecl-pagination__link"
          href="/example"
        >
          <span
            class="ecl-link__label"
          >
            25
          </span>
        </a>
      </li>
      <li
        class="ecl-pagination__item ecl-pagination__item--current"
      >
        <span
          aria-current="true"
          aria-label="Page 26"
          class="ecl-pagination__text ecl-pagination__text--summary"
        >
          26
        </span>
        <span
          aria-current="true"
          class="ecl-pagination__text ecl-pagination__text--full"
        >
          Page 26
        </span>
      </li>
      <li
        class="ecl-pagination__item"
      >
        <a
          aria-label="Go to page 27"
          class="ecl-link ecl-link--standalone ecl-pagination__link"
          href="/example"
        >
          <span
            class="ecl-link__label"
          >
            27
          </span>
        </a>
      </li>
      <li
        class="ecl-pagination__item"
      >
        <a
          aria-label="Go to page 28"
          class="ecl-link ecl-link--standalone ecl-pagination__link"
          href="/example"
        >
          <span
            class="ecl-link__label"
          >
            28
          </span>
        </a>
      </li>
      <li
        class="ecl-pagination__item ecl-pagination__item--next"
      >
        <a
          aria-label="Go to next page"
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-after ecl-pagination__link"
          href="/example"
        >
          <span
            class="ecl-link__label"
          >
            Next
          </span>
           
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs ecl-icon--rotate-90 ecl-link__icon"
            focusable="false"
          >
            <use
              xlink:href="/icons.svg#ui--corner-arrow"
            />
          </svg>
        </a>
      </li>
    </ul>
  </nav>
</test>
`;

exports[`EC - Pagination Default renders correctly with extra attributes 1`] = `
<test>
  <nav
    aria-label="Pagination"
    class="ecl-pagination"
    data-test="data-test-value"
    data-test-1="data-test-value-1"
  >
    <ul
      class="ecl-pagination__list"
    >
      <li
        class="ecl-pagination__item ecl-pagination__item--previous"
      >
        <a
          aria-label="Go to previous page"
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-pagination__link"
          href="/example"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs ecl-icon--rotate-270 ecl-link__icon"
            focusable="false"
          >
            <use
              xlink:href="/icons.svg#ui--corner-arrow"
            />
          </svg>
           
          <span
            class="ecl-link__label"
          >
            Previous
          </span>
        </a>
      </li>
      <li
        class="ecl-pagination__item"
      >
        <a
          aria-label="Go to page 24"
          class="ecl-link ecl-link--standalone ecl-pagination__link"
          href="/example"
        >
          <span
            class="ecl-link__label"
          >
            24
          </span>
        </a>
      </li>
      <li
        class="ecl-pagination__item"
      >
        <a
          aria-label="Go to page 25"
          class="ecl-link ecl-link--standalone ecl-pagination__link"
          href="/example"
        >
          <span
            class="ecl-link__label"
          >
            25
          </span>
        </a>
      </li>
      <li
        class="ecl-pagination__item ecl-pagination__item--current"
      >
        <span
          aria-current="true"
          aria-label="Page 26"
          class="ecl-pagination__text ecl-pagination__text--summary"
        >
          26
        </span>
        <span
          aria-current="true"
          class="ecl-pagination__text ecl-pagination__text--full"
        >
          Page 26
        </span>
      </li>
      <li
        class="ecl-pagination__item"
      >
        <a
          aria-label="Go to page 27"
          class="ecl-link ecl-link--standalone ecl-pagination__link"
          href="/example"
        >
          <span
            class="ecl-link__label"
          >
            27
          </span>
        </a>
      </li>
      <li
        class="ecl-pagination__item"
      >
        <a
          aria-label="Go to page 28"
          class="ecl-link ecl-link--standalone ecl-pagination__link"
          href="/example"
        >
          <span
            class="ecl-link__label"
          >
            28
          </span>
        </a>
      </li>
      <li
        class="ecl-pagination__item ecl-pagination__item--next"
      >
        <a
          aria-label="Go to next page"
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-after ecl-pagination__link"
          href="/example"
        >
          <span
            class="ecl-link__label"
          >
            Next
          </span>
           
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs ecl-icon--rotate-90 ecl-link__icon"
            focusable="false"
          >
            <use
              xlink:href="/icons.svg#ui--corner-arrow"
            />
          </svg>
        </a>
      </li>
    </ul>
  </nav>
</test>
`;

exports[`EC - Pagination Default renders correctly with extra class names 1`] = `
<test>
  <nav
    aria-label="Pagination"
    class="ecl-pagination custom-class custom-class--test"
  >
    <ul
      class="ecl-pagination__list"
    >
      <li
        class="ecl-pagination__item ecl-pagination__item--previous"
      >
        <a
          aria-label="Go to previous page"
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-pagination__link"
          href="/example"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs ecl-icon--rotate-270 ecl-link__icon"
            focusable="false"
          >
            <use
              xlink:href="/icons.svg#ui--corner-arrow"
            />
          </svg>
           
          <span
            class="ecl-link__label"
          >
            Previous
          </span>
        </a>
      </li>
      <li
        class="ecl-pagination__item"
      >
        <a
          aria-label="Go to page 24"
          class="ecl-link ecl-link--standalone ecl-pagination__link"
          href="/example"
        >
          <span
            class="ecl-link__label"
          >
            24
          </span>
        </a>
      </li>
      <li
        class="ecl-pagination__item"
      >
        <a
          aria-label="Go to page 25"
          class="ecl-link ecl-link--standalone ecl-pagination__link"
          href="/example"
        >
          <span
            class="ecl-link__label"
          >
            25
          </span>
        </a>
      </li>
      <li
        class="ecl-pagination__item ecl-pagination__item--current"
      >
        <span
          aria-current="true"
          aria-label="Page 26"
          class="ecl-pagination__text ecl-pagination__text--summary"
        >
          26
        </span>
        <span
          aria-current="true"
          class="ecl-pagination__text ecl-pagination__text--full"
        >
          Page 26
        </span>
      </li>
      <li
        class="ecl-pagination__item"
      >
        <a
          aria-label="Go to page 27"
          class="ecl-link ecl-link--standalone ecl-pagination__link"
          href="/example"
        >
          <span
            class="ecl-link__label"
          >
            27
          </span>
        </a>
      </li>
      <li
        class="ecl-pagination__item"
      >
        <a
          aria-label="Go to page 28"
          class="ecl-link ecl-link--standalone ecl-pagination__link"
          href="/example"
        >
          <span
            class="ecl-link__label"
          >
            28
          </span>
        </a>
      </li>
      <li
        class="ecl-pagination__item ecl-pagination__item--next"
      >
        <a
          aria-label="Go to next page"
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-after ecl-pagination__link"
          href="/example"
        >
          <span
            class="ecl-link__label"
          >
            Next
          </span>
           
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs ecl-icon--rotate-90 ecl-link__icon"
            focusable="false"
          >
            <use
              xlink:href="/icons.svg#ui--corner-arrow"
            />
          </svg>
        </a>
      </li>
    </ul>
  </nav>
</test>
`;
